frontend/packages/kubevirt-plugin/src/selectors/vm/selectors.ts

```diff
@@ -53,51 +55,30 @@ export const getVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[
 export const getDataVolumeTemplates = (vm: VMKind, defaultValue = []) =>
   _.get(vm, 'spec.dataVolumeTemplates') == null ? defaultValue : vm.spec.dataVolumeTemplates;
 
-export const getConfigMapVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
-  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('configMap'));
-
-export const getSecretVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
-  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('secret'));
-
-export const getServiceAccountVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
-  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('serviceAccount'));
-
-export const getEnvDiskVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] => [
-  ...getConfigMapVolumes(vm, defaultValue),
-  ...getSecretVolumes(vm, defaultValue),
-  ...getServiceAccountVolumes(vm, defaultValue),
-];
-
-export const getConfigMapDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter(
-    (disk) => !!getConfigMapVolumes(vm).find((vol) => vol.name === disk.name),
-  );
-
-export const getSecretDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter(
-    (disk) => !!getSecretVolumes(vm).find((vol) => vol.name === disk.name),
-  );
-
-export const getServiceAccountDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter(
-    (disk) => !!getServiceAccountVolumes(vm).find((vol) => vol.name === disk.name),
+const getVolumesByVolTypes = (
+  vm: VMKind,
+  volTypes: VolumeType[] = [],
+  defaultValue: V1Volume[] = [],
+): V1Volume[] =>
+  getVolumes(vm, defaultValue).filter((vol) =>
+    volTypes.some((volType) => new VolumeWrapper(vol).getType() === volType),
```

Comment:
we could use a Set here and do one `.has` call on the type.

Reply:
why is it better? isn't it the same complexity ?

Reply:
Depends on how you look at it. If you expect the volTypes to be always length of 3, then yes.

Anyway the main difference is that it would be nicer to read.